tests/api/test_layer_measurements.py

```diff
@@ -1,111 +1,105 @@
-from datetime import date
-
-import geopandas as gpd
-import numpy as np
+"""
+Test the Layer Measurement class
+"""
 import pytest
+import datetime
+
+from geoalchemy2.shape import to_shape
+from geoalchemy2.elements import WKTElement
 
 from snowexsql.api import LayerMeasurements
-from tests import DBConnection
+from snowexsql.tables import LayerData
 
 
-class TestLayerMeasurements(DBConnection):
-    """
-    Test the Layer Measurement class
-    """
-    CLZ = LayerMeasurements
+@pytest.fixture
+def layer_data(layer_data_factory, db_session):
+    layer_data_factory.create()
+    return db_session.query(LayerData).all()
 
-    def test_all_types(self, clz):
-        result = clz().all_types
-        assert result == ["density"]
+@pytest.mark.usefixtures("db_test_session")
+@pytest.mark.usefixtures("db_test_connection")
+@pytest.mark.usefixtures("layer_data")
+class TestLayerMeasurements:
+    @pytest.fixture(autouse=True)
+    def setup_method(self, layer_data):
+        self.subject = LayerMeasurements()
+        self.db_data = layer_data
 
-    def test_all_campaigns(self, clz):
-        result = clz().all_campaigns
-        assert result == ['Grand Mesa']
+    def test_all_campaigns(self):
+        result = self.subject.all_campaigns
+        assert result == [
+            record.site.campaign.name
+            for record in self.db_data
+        ]
+    
+    def test_all_observers(self):
+       result = self.subject.all_observers
+       assert result == [
+           observer.name
+           for record in self.db_data
+           for observer in record.site.observers
+       ]
 
-    def test_all_sites(self, clz):
-        result = clz().all_sites
-        assert result == ['Fakepit1']
+    def test_all_sites(self):
+        result = self.subject.all_sites
+        assert result == [
+            record.site.name
+            for record in self.db_data
+        ]
 
-    def test_all_dates(self, clz):
-        result = clz().all_dates
-        assert result == [date(2020, 1, 28)]
+    def test_all_dates(self):
+        result = self.subject.all_dates
+        assert result == [
+            record.site.date
+            for record in self.db_data
+        ]
 
-    def test_all_observers(self, clz):
-        result = clz().all_observers
-        assert result == ['TEST']
+@pytest.fixture
+def layer_data(layer_density_factory, db_session):
```

Comment:
Suggest of not re-using the same method name within a file and pick something unique (e.g. layer_data_with_density)

Reply:
Yeah I think that is a good call